src/write.c

```diff
@@ -746,6 +825,27 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         }
     }
 
+    // -----------------------------------------------------------------------
+    // Handle automatic tiling
+
+    encoder->data->tileRowsLog2 = encoder->tileRowsLog2;
+    encoder->data->tileColsLog2 = encoder->tileColsLog2;
+    if (encoder->autoTiling) {
+        // Use as many tiles as allowed by the minimum tile area requirement and impose a maximum
+        // of 8 tiles.
+        const int threads = 8;
```

Comment:
The maximum number of tiles is a hardcoded value of 8. This could be an encoder option. It should not be based on `encoder->maxThreads` alone because it also depends on how many worker threads the decoder is expected to have.

One solution is to change the `autoTiling` option to be an `int` and specify the maximum number of tiles. But then the `autoTiling` option won't be fully automatic.